src/hotspot/cpu/x86/assembler_x86.cpp

```diff
@@ -8973,9 +8979,10 @@ void Assembler::vinserti64x4(XMMRegister dst, XMMRegister nds, XMMRegister src,
   emit_int24(0x3A, (0xC0 | encode), imm8 & 0x01);
 }
 
-void Assembler::vinserti64x2(XMMRegister dst, XMMRegister nds, XMMRegister src, uint8_t imm8) {
+void Assembler::vinserti64x2(XMMRegister dst, XMMRegister nds, XMMRegister src, uint8_t imm8, int vector_len) {
    assert(VM_Version::supports_avx512dq(), "");
-   InstructionAttr attributes(AVX_512bit, /* vex_w */ true, /* legacy_mode */ false, /* no_mask_reg */ true, /* uses_vl */ true);
+   assert(vector_len == AVX_256bit || VM_Version::supports_avx512vl(), "");
```

Comment:
As this is an evex instruction, we could call it evinserti64x2. Also the assert should be: assert(vector_len == AVX_512bit || VM_Version::supports_avx512vl(), "");